bundles/org.openhab.binding.pilight/src/main/java/org/openhab/binding/pilight/internal/discovery/PilightBridgeDiscoveryService.java

```diff
@@ -78,52 +77,40 @@ protected void startScan() {
+                        final AtomicBoolean loop = new AtomicBoolean(true);
+                        while (loop.get()) {
                             DatagramPacket recvPack = new DatagramPacket(new byte[1024], 1024);
                             ssdp.receive(recvPack);
                             byte[] recvData = recvPack.getData();
-                            try (InputStreamReader recvInput = new InputStreamReader(new ByteArrayInputStream(recvData),
-                                    StandardCharsets.UTF_8)) {
-                                StringBuilder recvOutput = new StringBuilder();
-                                for (int value; (value = recvInput.read()) != -1;) {
-                                    recvOutput.append((char) value);
-                                }
-                                try (BufferedReader bufReader = new BufferedReader(
-                                        new StringReader(recvOutput.toString()))) {
-                                    Pattern pattern = Pattern.compile("Location:([0-9.]+):(.*)");
-                                    String line;
-                                    while ((line = bufReader.readLine()) != null) {
-                                        Matcher matcher = pattern.matcher(line);
-                                        if (matcher.matches()) {
-                                            String server = matcher.group(1);
-                                            Integer port = Integer.parseInt(matcher.group(2));
-                                            loop = false;
-                                            logger.debug("Found pilight daemon at {}:{}", server, port);
-
-                                            Map<String, Object> properties = new HashMap<>(2);
-                                            properties.put(PilightBindingConstants.PROPERTY_IP_ADDRESS, server);
-                                            properties.put(PilightBindingConstants.PROPERTY_PORT, port);
-
-                                            ThingUID uid = new ThingUID(PilightBindingConstants.THING_TYPE_BRIDGE,
-                                                    server.replace(".", "") + "" + port);
-
-                                            DiscoveryResult result = DiscoveryResultBuilder.create(uid)
-                                                    .withProperties(properties)
-                                                    .withLabel("Pilight Bridge (" + server + ")").build();
-
-                                            thingDiscovered(result);
-                                        }
-                                    }
-                                }
-                            }
+
+                            final Scanner scanner = new Scanner(new ByteArrayInputStream(recvData),
+                                    StandardCharsets.UTF_8);
+                            scanner.findAll("Location:([0-9.]+):(.*)").forEach(matchResult -> {
```

Comment:
You can avoid making `loop` an AtomicBoolean by using Stream.peek().count() instead of foreach. You can use the returned count value to set your `loop` value.
```suggestion
                            scanner.findAll("Location:([0-9.]+):(.*)").peek(matchResult -> {
```

Reply:
Good idea, but is it good practise to use `peek` for this? According to the documentation, it is mainly intended for debugging. 

> This method exists mainly to support debugging, where you want to see the elements as they flow past a certain point in a pipeline. 

(see https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-)